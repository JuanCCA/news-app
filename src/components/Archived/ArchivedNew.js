import React from 'react'
import Card from 'react-bootstrap/Card'
import {deleteArchived} from '../../helpers/Archived/deleteArchived'
import { formatDate } from '../../helpers/Utils/dateFormater'
import Button from 'react-bootstrap/Button'


const ArchivedNew = ({_id,title,description,date,content,author,archiveDate,setArchived}) => {
    
    
    const formatedDate = formatDate(date)
    // Delete an archived new from DB
    const handlerDeleteArchived = async() => {
        const resp = await deleteArchived(_id)
        if(resp === 200){
            handlerDeleteFromList()
        }
    }

    // Delete this item from the list 
    const handlerDeleteFromList = () => {
        setArchived( (news) => {
            console.log(news)
            return news.filter( doc => doc._id !== _id)
        })
    }


    return (
        <Card style={{ width: '90%' }} className="my-3">
        <Card.Body>
            <Card.Title>{title}</Card.Title>
            <Card.Subtitle className="mb-2 text-muted">{description}</Card.Subtitle>
            <Card.Text>{content}</Card.Text>
            <Card.Footer >

            <div className="row">
                <div className="col">
                    Archived date: {formatedDate} 
                </div>
                <div className="col">
                    Author: {author}
                </div>
                <div className="col">
                    <Button onClick={handlerDeleteArchived}
                            variant="dark">
                        <img 
                            src="./delete.png"
                            width="25"
                            height="25"
                            className="d-inline-block align-top"   
                            alt=""                 
                            >
                        </img>
                        Delete
                    </Button>
                </div>
            </div>
                
            </Card.Footer>
        </Card.Body>
        </Card>
    )
}

export default ArchivedNew
