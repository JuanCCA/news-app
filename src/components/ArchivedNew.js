import React from 'react'
import Card from 'react-bootstrap/Card'

const ArchivedNew = ({title,description,date,content,author,archiveDate}) => {
    return (
        <Card style={{ width: '90%' }} className="my-3">
        <Card.Body>
            <Card.Title>{title}</Card.Title>
            <Card.Subtitle className="mb-2 text-muted">{description}</Card.Subtitle>
            <Card.Text>{content}</Card.Text>
            <Card.Footer >

            <div className="row">
                <div className="col">
                    Date: {date} 
                </div>
                <div className="col">
                    Author: {author}
                </div>
                <div className="col">
                    <img 
                        role='button'
                        src="./delete.png"
                        width="30"
                        height="30"                    
                        //onClick={pape}
                        >
                    </img>
                </div>
            </div>
                
            </Card.Footer>
        </Card.Body>
        </Card>
    )
}

export default ArchivedNew
