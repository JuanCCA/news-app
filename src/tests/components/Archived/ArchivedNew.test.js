import { shallow } from 'enzyme'
import React from 'react'
import ArchivedNew from '../../../components/Archived/ArchivedNew'



describe('Testing component <ArchivedNew />', () => {
    
    const args = {
        _id:"28929837466959830853",
        title: "TITLE 1",
        description: "Description of the new",
        content: "content of the new",
        archiveDate: "2021-09-09T22:00:00.000+00:00",
        author: "Juan Carlos"
    }

    const setArchived = jest.fn()

    let wrapper = shallow(<ArchivedNew key={args._id}  {...args} setArchived={setArchived}/>)

    beforeEach( () => {
        jest.clearAllMocks();
        wrapper = shallow(<ArchivedNew key={args._id}  {...args} setArchived={setArchived}/>);
    });


    test('should create the component correctly with all of arguments needed', () => {
        
        expect(wrapper).toMatchSnapshot()


    })





})
