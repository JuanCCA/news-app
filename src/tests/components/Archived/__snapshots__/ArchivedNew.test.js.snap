// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing component <ArchivedNew /> should create the component correctly with all of arguments needed 1`] = `
<Card
  body={false}
  className="my-3"
  style={
    Object {
      "width": "90%",
    }
  }
>
  <CardBody>
    <CardTitle>
      TITLE 1
    </CardTitle>
    <CardSubtitle
      className="mb-2 text-muted"
    >
      Description of the new
    </CardSubtitle>
    <CardText>
      content of the new
    </CardText>
    <CardFooter>
      <div
        className="row"
      >
        <div
          className="col"
        >
          Archived date: 
          NaN/NaN/NaN  NaN:NaN
        </div>
        <div
          className="col"
        >
          Author: 
          Juan Carlos
        </div>
        <div
          className="col"
        >
          <Button
            active={false}
            disabled={false}
            onClick={[Function]}
            variant="dark"
          >
            <img
              alt=""
              className="d-inline-block align-top"
              height="25"
              src="./delete.png"
              width="25"
            />
            Delete
          </Button>
        </div>
      </div>
    </CardFooter>
  </CardBody>
</Card>
`;
